lib/activerecord-multi-tenant/query_rewriter.rb

```diff
@@ -324,6 +329,20 @@ def build_arel(*args)
 
       arel
     end
+
+    private
+    def relations_from_node_join(node_join)
```

Comment:
I think there is two places we can put this helper:

1) Right where you put it, inline in the module. This risks a potential conflict with internal Rails functions (I believe), but is straightfoward otherwise.

2) Dedicated helper module that gets called. That may also make this easier to test (since you could mock/create the arel nodes and write a unit test for the function)

I'd personally be fine with either, and in the interest of time might have gone for (1) myself. Though (2) feels like the slightly cleaner version.

Reply:
it does feel like 2 is a bit cleaner :) So I'll look into this option. Also there might be other functions that could be move to a helper module